z_infra_provisioning/cloud_infra_center/ocp_upi/tools/mirror-registry/01-mirror-registry.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -e
+
+# Variable Setup
+source $HOME/.bashrc
+
+LOCAL_REGISTRY_USERNAME="icic"
+LOCAL_REGISTRY_PASSWORD="icic"
+LOCAL_REGISTRY_HOSTNAME="image.registry.icic.ocp.com"
+LOCAL_REGISTRY_PORT="5008"
+VERSION="4.8.13"
+PULL_SECRET=""
+
+ARCH="s390x"
+
+if [ -z "$PULL_SECRET" ]
+then
+      echo "PULL_SECRET is empty" && exit 1
+fi
+
+# Adding registry to your pull-secret
+echo "*** Adding registry to your pull-secret ****"
+TOKEN=`echo -n $LOCAL_REGISTRY_USERNAME:$LOCAL_REGISTRY_PASSWORD  | base64 -w0`
+HOSTPORT=${LOCAL_REGISTRY_HOSTNAME}:${LOCAL_REGISTRY_PORT}
+REGISTRY_EMAIL="registry@${LOCAL_REGISTRY_HOSTNAME}"
+
+echo ${PULL_SECRET} >> ./pull-secret
+
+# Make a copy of your pull secret in JSON format:
+cat ./pull-secret | jq .  > ./pull-secret-dup.json
+
+# Add your credentials to  pull-secret.json
+cat pull-secret-dup.json | jq '.auths |= . +  {"HOSTPORT": { "auth": "TOKEN", "email": "you@example.com"}}' pull-secret-dup.json > pull-secret.json
+
+sed -i "s/TOKEN/$TOKEN/g" pull-secret.json
+sed -i "s/HOSTPORT/${HOSTPORT}/g" pull-secret.json
+sed -i "s/you@example.com/$REGISTRY_EMAIL/g" pull-secret.json
+
+BUILDNAME="ocp"
+#BUILDNUMBER="$(curl -s "https://mirror.openshift.com/pub/openshift-v4/clients/${BUILDNAME}/${VERSION}/release.txt" | grep 'Name:' | awk '{print $NF}')"
+BUILDNUMBER="$(curl -s "https://mirror.openshift.com/pub/openshift-v4/${ARCH}/clients/${BUILDNAME}/${VERSION}/release.txt" | grep 'Name:' | awk '{print $NF}')"
+
+if [ "$(echo "${BUILDNUMBER}" | cut -d '.' -f1-2)" == "4.2" ] && [ "$(echo "${BUILDNUMBER}" | cut -d '.' -f3)" -lt "14" ];then
+  OCP_RELEASE="${BUILDNUMBER}"
+else
+  OCP_RELEASE="${BUILDNUMBER}-${ARCH}"
+fi
+
+LOCAL_REGISTRY="${LOCAL_REGISTRY_HOSTNAME}:${LOCAL_REGISTRY_PORT}"
+LOCAL_REPOSITORY='icic/openshift4'
+#PRODUCT_REPO="$(curl -s "https://mirror.openshift.com/pub/openshift-v4/clients/${BUILDNAME}/${VERSION}/release.txt" | grep "Pull From:" | cut -d '/' -f2)"
+PRODUCT_REPO="$(curl -s "https://mirror.openshift.com/pub/openshift-v4/s390x/clients/${BUILDNAME}/${VERSION}/release.txt" | grep "Pull From:" | cut -d '/' -f2)"
+LOCAL_SECRET_JSON="./pull-secret.json"
+#RELEASE_NAME="$(curl -s "https://mirror.openshift.com/pub/openshift-v4/clients/${BUILDNAME}/${VERSION}/release.txt" | grep "Pull From:" | cut -d '/' -f3 | cut -d '@' -f1)"
+RELEASE_NAME="$(curl -s "https://mirror.openshift.com/pub/openshift-v4/s390x/clients/${BUILDNAME}/${VERSION}/release.txt" | grep "Pull From:" | cut -d '/' -f3 | cut -d '@' -f1)"
+
+# Mirroring Images
+echo "Mirroring Images..."
+echo "GODEBUG=x509ignoreCN=0 oc adm release mirror -a "${LOCAL_SECRET_JSON}" --from="quay.io/${PRODUCT_REPO}/${RELEASE_NAME}:${OCP_RELEASE}" --to-release-image="${LOCAL_REGISTRY}/${LOCAL_REPOSITORY}:${OCP_RELEASE}" --to="${LOCAL_REGISTRY}/${LOCAL_REPOSITORY}""
```

Comment:
one risk is the quay.io/openshift-release-dev/ocp-release tag may not have all releases.
You can add check by` curl -s https://quay.io/api/v1/repository/openshift-release-dev/ocp-release/tag/?onlyActiveTags=true | jq -r ".tags[].name"`